src/UI/Windows/Modals.cs

```diff
@@ -97,6 +98,44 @@ internal static bool GetArray(string windowTitle, string entryLabel, Type arrayE
         return false;
     }
 
+    internal static bool TryGetArray<T>(string windowTitle, string entryLabel, Array? initialValue, out Array? result)
```

Comment:
I don't like that there are now 2 copies of very similar code.  It is nice to have generic method but needs refactoring so that there are as few duplicated lines as possible.

Simplest way to do that would be for the generic to call the specific e.g.

```csharp
internal static bool TryGetArray<T>(string windowTitle, string entryLabel, Array? initialValue, out Array? result)
{
    return GetArray(windowTitle, entryLabel, typeof(T), initialValue, out result);
}
```

Also lets rename both to be called `TryGetArray`

Reply:
A agree completely.

However, it will require a LOT of work, as that will require a bunch of non-generic code to be genericized.

So, since the API isn't really intended for external consumption, I thought it would be acceptable to do it in a more gradual way. During that transition, there would be both, but the end goal is no duplication, for sure. Does that make sense?

Reply:
Just to add to my previous response, the current implementation limits just how far I can take generics, since most of the rest isn't generic and passes Types around. That approach is actually somewhat _easier_, ironically, around some of the reflection, but of course has its costs, or at least can cause at least a little bit of code duplication, though most or all of that can be avoided by using proxies that do the Type-based checks and then call the generic for the actual implementation. But, it makes testing more difficult, and the reflection wouldn't be necessary with a pure generic implementation, anyway.

I've got a bunch of other thoughts, too - all of which tie in - which could have profound impacts on the ease of maintaining and improving TGD, in the future, but they're all definitely their own work items.